sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c

```diff
@@ -115,15 +115,15 @@ static az_result root(az_event_policy* me, az_event event)
  * and passes the event for the ready state to handle
  */
 AZ_INLINE az_result send_to_ready_if_topic_matches(
-    az_mqtt5_rpc_client_policy* this_policy,
+    az_mqtt5_rpc_client* this_policy,
     az_event event,
     az_event_policy_handler source_state)
 {
   az_mqtt5_recv_data* recv_data = (az_mqtt5_recv_data*)event.data;
 
   // Ensure pub is of the right topic
   if (az_span_topic_matches_sub(
```

Comment:
As mentioned in a previous PR, these functions should be part of the rpc_client_codec instead of requiring access to internals.